src/target/puya.c

```diff
@@ -114,39 +121,83 @@ bool puya_probe(target_s *target)
 	size_t flash_size = 0U;
 
 	const uint32_t dbg_idcode = target_mem32_read32(target, PUYA_DBG_IDCODE);
-	if ((dbg_idcode & 0xfffU) == 0) {
-		const uint32_t flash_ram_sz = target_mem32_read32(target, PUYA_FLASH_RAM_SZ);
+	if (dbg_idcode == 0x06188061U) {
```

Comment:
Perhaps define this constant as `#define ID_PY32F07x 0x06188061U`? that way it's more self-explanatory.